backend/app/api/auth.py

```
@@ -60,35 +70,48 @@ async def register_user(user_data: UserRegister, db: Session = Depends(get_db)):
+        db, user_credentials.email, user_credentials.password
+    )
     if not user:
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED,
             detail="Incorrect email or password",
             headers={"WWW-Authenticate": "Bearer"},
         )
 
-    if not user.is_active:
+    # Type guard: user is now guaranteed to be User instance
+    assert isinstance(user, User)
```

Ruff (0.12.2)

Use of `assert` detected

(S101)